tools/benchmark/binsize/src/main.rs

```diff
@@ -0,0 +1,26 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::fs;
+
+fn wasm_filesize(dir: &str) -> Result<u64, std::io::Error> {
+    let paths = fs::read_dir(dir).expect("Directory wasm/ not found!");
+    for path in paths {
+        let p = path.unwrap().path();
+        if let Some(suffix) = p.extension() {
+            if suffix == "wasm" {
+                println!(
+                    "{{\"biggerIsBetter\":false,\"name\":{:?},\"unit\":\"bytes\",\"value\":{}}}",
+                    p.file_stem().unwrap(),
+                    p.metadata()?.len()
+                );
+            }
+        }
+    }
+    Ok(1)
+}
+
+fn main() {
+    let _rc = wasm_filesize("wasmpkg");
```

Comment:
Question: Is this path correct? I see a `wasm-opt` directory, and it seems like we should track optimized code as the binsize metric.

```
~/dev/icu4x/wasmpkg on icu4x/main at 16:00:20
➤ tree
.
├── borrowed_pattern.d
├── borrowed_pattern.wasm
├── code_line_diff.d
├── code_line_diff.wasm
├── derive.d
├── derive.wasm
├── elevator_floors.d
├── elevator_floors.wasm
├── filter_langids.d
├── filter_langids.wasm
├── language_names_hash_map.d
├── language_names_hash_map.wasm
├── language_names_lite_map.d
├── language_names_lite_map.wasm
├── owned_pattern.d
├── owned_pattern.wasm
├── permyriad.d
├── permyriad.wasm
├── simple.d
├── simple.wasm
├── syntatically_canonicalize_locales.d
├── syntatically_canonicalize_locales.wasm
├── tui.d
├── tui.wasm
├── twiggy
├── unicode_bmp_blocks_selector.d
├── unicode_bmp_blocks_selector.wasm
├── unread_emails.d
├── unread_emails.wasm
├── wasm-decompile
│   ├── borrowed_pattern.dcmp
│   ├── code_line_diff.dcmp
│   ├── derive.dcmp
│   ├── elevator_floors.dcmp
│   ├── filter_langids.dcmp
│   ├── language_names_hash_map.dcmp
│   ├── language_names_lite_map.dcmp
│   ├── owned_pattern.dcmp
│   ├── permyriad.dcmp
│   ├── simple.dcmp
│   ├── syntatically_canonicalize_locales.dcmp
│   ├── tui.dcmp
│   ├── unicode_bmp_blocks_selector.dcmp
│   ├── unread_emails.dcmp
│   ├── work_log.dcmp
│   └── writeable_message.dcmp
├── wasm-opt
│   ├── borrowed_pattern+opt.wasm
│   ├── code_line_diff+opt.wasm
│   ├── derive+opt.wasm
│   ├── elevator_floors+opt.wasm
│   ├── filter_langids+opt.wasm
│   ├── language_names_hash_map+opt.wasm
│   ├── language_names_lite_map+opt.wasm
│   ├── owned_pattern+opt.wasm
│   ├── permyriad+opt.wasm
│   ├── simple+opt.wasm
│   ├── syntatically_canonicalize_locales+opt.wasm
│   ├── tui+opt.wasm
│   ├── unicode_bmp_blocks_selector+opt.wasm
│   ├── unread_emails+opt.wasm
│   ├── work_log+opt.wasm
│   └── writeable_message+opt.wasm
├── wat
│   ├── borrowed_pattern.wat
│   ├── code_line_diff.wat
│   ├── derive.wat
│   ├── elevator_floors.wat
│   ├── filter_langids.wat
│   ├── language_names_hash_map.wat
│   ├── language_names_lite_map.wat
│   ├── owned_pattern.wat
│   ├── permyriad.wat
│   ├── simple.wat
│   ├── syntatically_canonicalize_locales.wat
│   ├── tui.wat
│   ├── unicode_bmp_blocks_selector.wat
│   ├── unread_emails.wat
│   ├── work_log.wat
│   └── writeable_message.wat
├── work_log.d
├── work_log.wasm
├── writeable_message.d
└── writeable_message.wasm
```

The binsize difference is pretty substantial.

Reply:
Changed the path to wasmpkg/wasm-opt. The optimized binaries may not contain the additional information required for analyzing and debugging size regression, though.